### PR TITLE
Data saves to backend

### PR DESCRIPTION
**Pull request by MSeechan**
User input data from the contact page is now saved to the backend. I also updated the converToJson() to convertToJsonUsingGson() and all files were auto-formatted.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 1]
```diff
@@ -1,26 +1,59 @@
...
         String emailValue = request.getParameter("contact-email");
         String messageValue = request.getParameter("contact-message");
 
-        // Print the value so you can see it in the server logs.
-        System.out.println("You submitted: " + nameValue + " |Email: " + emailValue + " |Message: " + messageValue);
+        String timeStamp = LocalTime.now(ZoneId.of("America/Los_Angeles")).truncatedTo(ChronoUnit.SECONDS).toString();
+
+        // clean inputs
+        if (nameValue != null) {
+            nameValue = Jsoup.clean(nameValue, Whitelist.none());
```
Comment (andy-keene) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:37]: Good security! 👓

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 2]
```diff
@@ -1,26 +1,59 @@
...
+        if (messageValue != null) {
+            messageValue = Jsoup.clean(messageValue, Whitelist.none());
+        }
+
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+        KeyFactory keyFactory = datastore.newKeyFactory().setKind("Contact");
+        FullEntity contactEntity = Entity.newBuilder(keyFactory.newKey()).set("contact-name", nameValue)
+                .set("contact-email", emailValue).set("contact-message", messageValue).set("timestamp", timeStamp)
+                .build();
+        datastore.put(contactEntity);
```
Comment (andy-keene) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:51]: Have you been able to confirm this data is being saved?

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java [hunk 3]
```diff
@@ -1,26 +1,59 @@
...
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
 @WebServlet("/form-handler")
 public class FormHandlerServlet extends HttpServlet {
```
Comment (andy-keene) [inline portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:24]: optional: What about a quick doGet() to retrieve all contact info?
```

  @Override
  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
    Query<Entity> query =
        Query.newEntityQueryBuilder().setKind("Contact").setOrderBy(OrderBy.desc("timeStamp")).build();
    QueryResults<Entity> results = datastore.run(query);

    List<Contact> contacts = new ArrayList<>();

    // todo(mseechan): Translate each data store entity to a Java object. 
    // I'll leave this to you, but the ListTasksServlet is a great example 

    Gson gson = new Gson();

    response.setContentType("application/json;");
    response.getWriter().println(gson.toJson(contacts));
  }
```

**Review by andy-keene — APPROVED** [review]
✅  LGTM! (with an optional stretch)

Happy to see you got this up and running. This will be super valuable knowledge for the team project.

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java (hunk 2 above)
Comment (MSeechan) [reply portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:51]: Yes, I checked on the gcloud datastore admin page link provided by the tutorial.

File: portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java (hunk 3 above)
Comment (MSeechan) [reply portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java:24]: Thanks! I'm currently reworking the example servlet to fit my page. I'll let you know if I run into any issues.

**Pull request merged**